corehq/apps/userreports/reports/view.py

```diff
@@ -78,12 +93,28 @@ def has_permissions(self, domain, user):
         return True
 
     def get_context_data(self, **kwargs):
-        return {
+        context = {
             'domain': self.domain,
             'report': self,
             'filter_context': self.filter_context,
-            'url': reverse(self.slug, args=[self.domain, self.report_config_id]),
-            'headers': self.headers,
+            'url': self.url,
+            'headers': self.headers
+        }
+        context.update(self.saved_report_context_data)
+        return context
+
+    @property
+    def saved_report_context_data(self):
+        current_config_id = self.request.GET.get('config_id')
+        return {
+            'report_configs': ReportConfig.by_domain_and_owner(
+                self.domain, self.request.couch_user._id, report_slug=self.slug
+            ),
```

Comment:
flagging that Arya is working on language for a warning about sending
patient-level data

On Wed, Mar 18, 2015 at 12:05 PM, Nick Pellegrino <notifications@github.com>
wrote:

> In corehq/apps/userreports/reports/view.py
> <https://github.com/dimagi/commcare-hq/pull/5886#discussion_r26677669>:
>
> >              'domain': self.domain,
> >              'report': self,
> >              'filter_context': self.filter_context,
> > -            'url': reverse(self.slug, args=[self.domain, self.report_config_id]),
> > -            'headers': self.headers,
> > +            'url': self.url,
> > +            'headers': self.headers
> > +        }
> > +        context.update(self.saved_report_context_data)
> > +        return context
> > +
> > +    @property
> > +    def saved_report_context_data(self):
> > +        current_config_id = self.request.GET.get('config_id')
> > +        return {
> > +            'report_configs': ReportConfig.by_domain_and_owner(
> > +                self.domain, self.request.couch_user._id, report_slug=self.slug
> > +            ),
>
> We could do something more complex like only load it when "Favorites" is
> clicked, but I'd rather put off that optimization until it's needed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/5886/files#r26677669>.
>